### DISCUSSION TITLE
Questions about natural boundary condition and non-zero normal stress in INSAD

### DISCUSSION THREAD

**Discussion by doquang** [idaholab/moose#26826] [Q&A Modules: Navier-Stokes]
Hi,

I am trying to extend the INSAD module to include natural boundary conditions for the mass equation and non-zero normal stress for the momentum equation. See https://github.com/doquang/slip_flow. It seems to work for the low Re case (rho=400), but doesn't converge for the rho=1000 case. I'm wondering,

- Is the way I'm doing it correct?
- If correct, is there any way to improve convergence?
- For the working cases (rho=400). I still see some normal flow through the surface `100`. Is that acceptable?

Thanks!
![image](https://github.com/idaholab/moose/assets/3186923/718ddee6-3878-4fc8-a1b8-c7515dcf71c4)

**Comment by lindsayad** [comment]
I don't know what surface 100 is, but these results look reasonable to me with an initial eye test. If you submit a pull request to MOOSE I will give it a detailed look-over.

I am not surprised that you are having problems converging with a Reynolds number of 1000. You probably need more diffusion which is typically achieved by adding a turbulence model

**Reply by doquang** [reply]
@lindsayad - Please see the PR ##26827

**Reply by doquang** [reply]
@lindsayad - I have to create a new PR #26833
